README
NAME
    Module::Cloud - Generates a tag cloud for modules used in given code

VERSION
    version 1.100860

SYNOPSIS
      use Module::Cloud;
      my @dirs = qw(foo bar baz);
      my $clouder = Module::Cloud->new(dir => \@dirs);
      my $cloud = $clouder->get_cloud;  # returns an HTML::TagCloud object
      print $cloud->html_and_css;

DESCRIPTION
    This class traverses the given directories, searches for perl code,
    extracts the modules used by this code and generates a HTML::TagCloud
    object that gives an impression of how often each module is used.

METHODS
  get_cloud
      my $cloud = $clouder->get_cloud;

    Traverses the directories set on the "dir" accessor and searches for
    files containing perl code. For each such file, the modules it uses are
    extracted. From this data a HTML::TagCloud object is created and
    returned. The more often a module is used the bigger it will appear in
    the tag cloud.

INSTALLATION
    See perlmodinstall for information and options on installing Perl
    modules.

BUGS AND LIMITATIONS
    No bugs have been reported.

    Please report any bugs or feature requests through the web interface at
    <http://rt.cpan.org/Public/Dist/Display.html?Name=Module-Cloud>.

AVAILABILITY
    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit <http://www.perl.com/CPAN/> to find a
    CPAN site near you, or see <http://search.cpan.org/dist/Module-Cloud/>.

    The development version lives at
    <http://github.com/hanekomu/Module-Cloud/>. Instead of sending patches,
    please fork this project using the standard git and github
    infrastructure.

AUTHOR
      Marcel Gruenauer <marcel@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2007 by Marcel Gruenauer.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Perl extension Module-Cloud

1.100860  2010-03-27 12:54:56 Europe/Vienna
    - converted the distribution to Dist::Zilla-style

0.10  Thu Jul 25 15:27:43 CEST 2008 (Marcel Gruenauer <marcel@cpan.org>)
    - Changed dist style and Changes back to standard. People didn't like it -
      the nail that sticks out gets hammered down.
    - Added standard test files; this will also help with CPANTS.

0.09  Thu, 29 May 2008 12:06:01 -0000 (Marcel Gruenauer <marcel@cpan.org>)
    - Converted Changes file to YAML style
    - .shipit: added Twitter step
    - Makefile.PL: added auto_install() and process_templates()
    - lib/*: converted to template
    - updated MANIFEST
    - added t/perlcriticrc
    - updated MANIFEST
    - moved generated documentation of interited methods to after documentation
      of this class' methods so the manpage is more readable
    - .shipit: fixed svk.tagpattern
    - tags: NEWFEATURE

0.08  Sat, 03 Nov 2007 17:26:44 +0100 (Marcel Gruenauer <marcel@cpan.org>)
    - exclude pod coverage tests

0.07  Sat, 20 Oct 2007 22:32:19 +0200 (Marcel Gruenauer <marcel@cpan.org>)
    - simplified accessor declaration

0.06  Thu, 18 Oct 2007 10:09:39 +0200 (Marcel Gruenauer <marcel@cpan.org>)
    - fixed version requirement of Test::More

0.05  Sun, 23 Sep 2007 11:41:07 +0200 (Marcel Gruenauer <marcel@cpan.org>)
    - fixed documentation
    - now uses Module::Install::StandardTests

0.04  Fri, 07 Sep 2007 21:53:41 +0200 (Marcel Gruenauer <marcel@cpan.org>)
    - keep shipit happy (actually, fixed MANIFEST)

0.03  Fri, 07 Sep 2007 22:00:45 +0200 (Marcel Gruenauer <marcel@cpan.org>)
    - added t/01_this_dist_cloud.t
    - fixed some bugs

0.02  Fri, 07 Sep 2007 21:18:42 +0200 (Marcel Gruenauer <marcel@cpan.org>)
    - added boilerplate code, getting ready to ship

0.01  Tue, 08 May 2007 04:05:59 +0100 (Marcel Gruenauer <marcel@cpan.org>)
    - original version


lib/Module/Cloud.pm
use 5.008;
use strict;
use warnings;

package Module::Cloud;
our $VERSION = '1.100860';
# ABSTRACT: Generates a tag cloud for modules used in given code
use HTML::TagCloud;
use File::Find::Rule::MMagic;
use Module::ExtractUse;
use parent 'Class::Accessor::Complex';
__PACKAGE__
    ->mk_new
    ->mk_array_accessors('dir');

sub get_cloud {
    my $self = shift;
    my @text_files;
    for my $dir ($self->dir) {
        push @text_files, grep { !/\/Makefile.PL$/ }
          grep { !/\.t$/ }
          grep { !/\/\.svn\// }
          find(file => magic => [qw(text/* x-system/*)], in => $dir);
    }
    my $extractor = Module::ExtractUse->new;
    my %modules;
    for my $textfile (@text_files) {
        my $used = $extractor->extract_use($textfile);
        $modules{$_}++ for keys %{ $used->{found} };
    }
    my $cloud = HTML::TagCloud->new;
    while (my ($module, $count) = each %modules) {
        $cloud->add($module, "http://search.cpan.org/search?query=$module",
            $count);
    }
    $cloud;
}
1;


__END__
=pod

=head1 NAME

Module::Cloud - Generates a tag cloud for modules used in given code

=head1 VERSION

version 1.100860

=head1 SYNOPSIS

  use Module::Cloud;
  my @dirs = qw(foo bar baz);
  my $clouder = Module::Cloud->new(dir => \@dirs);
  my $cloud = $clouder->get_cloud;  # returns an HTML::TagCloud object
  print $cloud->html_and_css;

=head1 DESCRIPTION

This class traverses the given directories, searches for perl code, extracts
the modules used by this code and generates a L<HTML::TagCloud> object that
gives an impression of how often each module is used.

=head1 METHODS

=head2 get_cloud

  my $cloud = $clouder->get_cloud;

Traverses the directories set on the C<dir> accessor and searches for files
containing perl code. For each such file, the modules it uses are extracted.
From this data a L<HTML::TagCloud> object is created and returned. The more
often a module is used the bigger it will appear in the tag cloud.

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org/Public/Dist/Display.html?Name=Module-Cloud>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see
L<http://search.cpan.org/dist/Module-Cloud/>.

The development version lives at
L<http://github.com/hanekomu/Module-Cloud/>.
Instead of sending patches, please fork this project using the standard git
and github infrastructure.

=head1 AUTHOR

  Marcel Gruenauer <marcel@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2007 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



Makefile.PL

use strict;
use warnings;

 BEGIN { require 5.008; } 

use ExtUtils::MakeMaker 6.11;



my %WriteMakefileArgs = (
                       'test' => {
                                   'TESTS' => 't/*.t'
                                 },
                       'NAME' => 'Module::Cloud',
                       'DISTNAME' => 'Module-Cloud',
                       'CONFIGURE_REQUIRES' => {
                                                 'ExtUtils::MakeMaker' => '6.11'
                                               },
                       'AUTHOR' => 'Marcel Gruenauer <marcel@cpan.org>',
                       'BUILD_REQUIRES' => {},
                       'ABSTRACT' => 'Generates a tag cloud for modules used in given code',
                       'EXE_FILES' => [
                                        'bin/module_cloud'
                                      ],
                       'VERSION' => '1.100860',
                       'PREREQ_PM' => {
                                        'Class::Accessor::Complex' => '0',
                                        'Scalar::Util' => '0',
                                        'parent' => '0',
                                        'English' => '0',
                                        'FindBin' => '0',
                                        'HTML::TagCloud' => '0',
                                        'Module::ExtractUse' => '0',
                                        'File::Find::Rule::MMagic' => '0',
                                        'Carp' => '0',
                                        'Test::More' => '0.88',
                                        'File::Find' => '0',
                                        'File::Temp' => '0'
                                      },
                       'LICENSE' => 'perl'
                     );


delete $WriteMakefileArgs{LICENSE}
  unless eval { ExtUtils::MakeMaker->VERSION(6.31) };

WriteMakefile(%WriteMakefileArgs);



